src/mono/wasm/test-main.js

```diff
@@ -158,6 +159,7 @@ function initRunArgs() {
     runArgs.enableGC = runArgs.enableGC === undefined ? true : runArgs.enableGC;
     runArgs.diagnosticTracing = runArgs.diagnosticTracing === undefined ? false : runArgs.diagnosticTracing;
     runArgs.debugging = runArgs.debugging === undefined ? false : runArgs.debugging;
+    runArgs.deepWorkDir = runArgs.deepWorkDir === undefined ? false : runArgs.deepWorkDir;
```

Comment:
I think we will re-visit `runArgs` in near future, so I'm hesitant to add more to it. 
Is this just to keep the behaior backward compatible in some scenario ? 
Which scenarios would be affected if it was true by default ?

Reply:
If the default value is true, the test running on the V8 engine will search deeper than the PATH where the file actually exists when reading dotnet.wasm and the test will fail.
Therefore, this will affect all tests that test with V8.

This value is only used in the RunOutOfAppBundle test, but a change in the default value would have many affects...

Reply:
Could that be fixed only for V8 inside `dotnet.js` without making it configurable ?

Reply:
If it is possible to overwrite `scriptDirectory` with `import.meta.url` before loading dotnet.wasm in https://github.com/dotnet/runtime/blob/main/src/mono/wasm/runtime/es6/dotnet.es6.pre.js , for example, you may not need to perform this hack.
However, this will not work if the emscripten initialization script overwrites it with `var scriptDirectory` or something similar.

Reply:
Maybe we could install better default `Module.locateFile` some merge of your 
`scriptDirectory = import.meta.url.slice(0, import.meta.url.lastIndexOf('/')) + '/';` with `test-main.js` 
```
let toAbsoluteUrl = function (path, prefix) {
    if (prefix.startsWith("/")) {
        return path;
    }
    return prefix + path;
}
if (is_browser) {
    const anchorTagForAbsoluteUrlConversions = document.createElement('a');
    toAbsoluteUrl = function toAbsoluteUrl(path, prefix) {
        anchorTagForAbsoluteUrlConversions.href = prefix + path;
        return anchorTagForAbsoluteUrlConversions.href;
    }
}
```

And do it in https://github.com/dotnet/runtime/blob/6f7064dc170a46d1a5e67823f5178b8862b27a58/src/mono/wasm/runtime/exports.ts#L250

only if there is no user override